xwrf/accessors.py

```diff
@@ -27,6 +30,47 @@ def __init__(self, xarray_obj: xr.Dataset | xr.DataArray) -> WRFAccessor:
 class WRFDataArrayAccessor(WRFAccessor):
     """Adds a number of WRF specific methods to xarray.DataArray objects."""
 
+    def destagger(
+        self, stagger_dim: str | None = None, unstaggered_dim_name: str | None = None
+    ) -> xr.DataArray:
+        """
+        Destagger a single WRF xarray.DataArray
+
+        Parameters
+        ----------
+        stagger_dim : str, optional
+            Name of dimension to unstagger. Defaults to guessing based on name (ends in "_stag")
+        unstaggered_dim_name : str, option
+            String to which to rename the dimension after destaggering. Example would be
+            "west_east" for "west_east_stag". By default the dimenions will be renamed the text in
+            front of "_stag" from the "stagger_dim" field.
+
+        Returns
+        -------
+        xarray.DataArray
+            The destaggered DataArray with renamed dimension and adjusted coordinates.
+        """
+        new_variable = _destag_variable(
+            self.xarray_obj.variable, stagger_dim=stagger_dim, unstag_dim_name=unstaggered_dim_name
+        )
+
+        # Need to recalculate staggered coordinates, as they don't already exist independently
+        # in a DataArray context
+        new_coords = {}
+        for coord_name, coord_data in self.xarray_obj.coords.items():
+            if set(coord_data.dims).difference(set(new_variable.dims)):
+                # Has a dimension not in the destaggered output (and so still staggered)
+                new_name = _rename_staggered_coordinate(
+                    coord_name, stagger_dim=stagger_dim, unstag_dim_name=unstaggered_dim_name
+                )
+                new_coords[new_name] = _destag_variable(
+                    coord_data, stagger_dim=stagger_dim, unstag_dim_name=unstaggered_dim_name
+                )
```

Comment:
I'm not sure how significant this problem is and please correct me if I'm wrong, but aren't we making some error in destaggering lat/long coordinates which are in some projections on curvilinear grids? Would it not be correct to destagger `x`, `y`, `z` and then compute the `lat`/`long` coords using the `wrf_transformation` object?

Reply:
I'd have to go back and check the WRF code itself to see how it computes XLAT vs. XLAT_U/XLAT_V to know for sure. Regardless, I would think that error from "midpoint-style" interpolation versus a full projection-aware approach technically shows up in any other field as well, and is typically not considered an issue there either. 

I don't think it would be worth it to do a full re-computation, but perhaps we should make this clear in documentation (so that a user would know to trust the x/y/z coordinates or the XLAT/XLONG from the full dataset over those in this DataArray)?

Reply:
Well, afaik the difference is that any other field is defined on the WRF-grid which is rectilinear while the `XLAT` and `XLONG` values are the only ones native on the curvilinear grid. But upon thinking about it a bit more, I'm not even sure doing the transformation correctly is possible, since all needed data to set up the `pyproj` object is in the global attributes rather than the variable attributes. So I guess adding a huge red warning to the docs is the best way to go.

I'm not really sure how the relationship between `Dataset` and `DataArray` coordinates is, but if they keep a copy it would maybe be smart to add a `no_coords` kwarg to `WRFDataArrayAccessor.destagger`, and just copy the global coords into the variables in `WRFDatasetAccessor.destagger`.

Reply:
I've added an `exclude_staggered_auxiliary_coords` option and commentary about staggered auxiliary coordinate handling to the DataArray method, as well as updated the Dataset method's Notes to more directly explain what happens with coordinates. Hopefully that resolves everything here!